core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -69,12 +73,11 @@ private[spark] class SparkUpgradeException(version: String, message: String, cau
 /**
  * Arithmetic exception thrown from Spark with an error class.
  */
-class SparkArithmeticException(
-    val errorClass: Option[String],
-    val messageParameters: Seq[String])
-  extends ArithmeticException(SparkError.getMessage(errorClass.get, messageParameters))
-    with SparkError {
+class SparkArithmeticException(errorClass: String, messageParameters: Array[String])
+  extends ArithmeticException(SparkThrowableHelper.getMessage(errorClass, messageParameters))
+    with SparkThrowable {
 
-  def this(errorClass: String, messageParameters: Seq[String]) =
-    this(errorClass = Some(errorClass), messageParameters = messageParameters)
+  override def getErrorClass: String = errorClass
+  override def getMessageParameters: Array[String] = messageParameters
```

Comment:
It seems like we are totally fine if `SparkThrowable` does not have `getMessageParameters`. I think the message parameter is an internal thing and we shouldn't expose it to the end-users via a public API.

Reply:
Good call - this was in order to construct a `ParseException` from an `AnalysisException`, but I was able to work around this by making `messageParameters` accessible as a `val`.